packages/k8s.contrib.crd/internal/ModuleGenerator.pkl

```diff
@@ -202,29 +209,26 @@ local function getDocComment(schema: JsonSchema|Boolean, type: "class"|"module"|
     null
   else
     let (docCommentText: String =
-      List(
-        schema.description,
-        if (type is "module") "This module was generated from the CustomResourceDefinition at <\(baseUri)>." else null,
-        if (type is "class"|"module") getWarnings(schema, type) else null,
-        if (schema.default != null) "Default if undefined: `\(pcfRenderer.renderValue(schema.default))`" else null
-      )
-        .filterNonNull()
-        .join("\n\n")
+      new Listing {
+        schema.description
+        when (type is "module") { "This module was generated from the CustomResourceDefinition at <\(baseUri)>." }
+        when (type is "class"|"module") { getWarnings(schema, type) }
+        when (schema.default != null) { "Default if undefined: `\(pcfRenderer.renderValue(schema.default))`" }
+      }.toList().filterNonNull().join("\n\n")
     )
       if (docCommentText.isEmpty) null
-      else new DocCommentNode {
-        value = docCommentText
-        autoWrap = true
-      }
+      else
+        new DocCommentNode {
+          value = docCommentText
+          autoWrap = true
+        }
 
 local function getWarnings(schema: JsonSchema, type: "class"|"module"): String? =
   if (type == "class") null
-  else if (schema.type == "object")
-    "WARN: The root schema describes open-ended properties, but this is not possible to describe at the module level."
```

Comment:
Kubernetes apparently differs from OpenAPI _and_ JSON Schema because `properties` means a closed struct, and `additionalProperties` means arbitrary keys, and there's no in-between. So, this warning isn't very meaningful.